File: LeetCode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cs
public class MyQueue {
    private readonly Stack<int> s_in = new Stack<int>();
    private readonly Stack<int> s_out = new Stack<int>();
    
    public void Push(int x) {
        s_in.Push(x);
    }
    
    public int Pop() {
        EnsureOutputStackHasElements();
        return s_out.Pop();
    }

    
    public int Peek() {
        EnsureOutputStackHasElements();
        return s_out.Peek();
    }
    
    public bool Empty() {
        return s_in.Count == 0 && s_out.Count == 0;
    }

    private void EnsureOutputStackHasElements()
    {
        if (s_out.Count == 0)
        {
            while (s_in.Count > 0)
            {
                s_out.Push(s_in.Pop());
            }
        }
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.Push(x);
 * int param_2 = obj.Pop();
 * int param_3 = obj.Peek();
 * bool param_4 = obj.Empty();
 */

File: LeetCode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.inbound = []
        self.outbound = []

    def _pour_into_outboud(self) -> None:
        while self.inbound:
            self.outbound.append(self.inbound.pop())

    def push(self, x: int) -> None:
        self.inbound.append(x)

    def pop(self) -> int:
        if not self.outbound:
            self._pour_into_outboud()

        return self.outbound.pop()

    def peek(self) -> int:
        if not self.outbound:
            self._pour_into_outboud()
        return self.outbound[-1]

    def empty(self) -> bool:
        return not self.inbound and not self.outbound


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()